src/main.nr

```diff
@@ -7,7 +9,16 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
-    
+    let size = 2.pow_32(32) as u64;
+    let mut hash_value : u64 = 0x811C9DC5;
+    let bytes = u2b::u32_to_u8(num);
+
+    for i in bytes {
+        let product = (hash_value * (PRIME as u64)) % size;
```

Comment:
You can try doing something like below to avoid having to do the % operation as Noir itself does % in case of overflow

fn calculate_hash(num: u32) -> u32 {

    let size = 2.pow_32(32);
    let mut hash_value = 0x811C9DC5;
    let bytes = u2b::u32_to_u8(num);

    for i in bytes {
        let product = (hash_value * PRIME);
        hash_value = product ^ i as u32;
    };

    hash_value
}